modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/ExtensionComponent.java

```diff
@@ -252,23 +283,27 @@ private void initConfigurationResolver() {
 
     if (isConfigurationSpecified()) {
       // the config is dynamic
-      configurationResolver = event -> {
-        ConfigurationInstance instance = configurationProvider.get().get(event);
-        if (instance == null) {
-          throw new IllegalModelDefinitionException(format(
-                                                           "Root component '%s' contains a reference to config '%s' but it doesn't exists",
-                                                           getLocation().getRootContainerName(),
-                                                           configurationProvider));
-        }
-
-        return of(instance);
-      };
+      // the optional should be always present at this point because we already checked dependsOnEvent
+      configurationResolver = event -> resolveConfigFromProvider(getConfigurationProvider().get(), event);
     } else {
       // obtain implicit instance
       configurationResolver = event -> extensionManager.getConfiguration(extensionModel, componentModel, event);
     }
   }
 
+  private Optional<ConfigurationInstance> resolveConfigFromProvider(ConfigurationProvider configurationProvider,
+                                                                    CoreEvent event) {
+    ConfigurationInstance instance = configurationProvider.get(event);
+    if (instance == null) {
```

Comment:
A configProvider for a static config will never return null. If the configProvider points to an expression, then the error message doesn't make sense

Reply:
what I meant to say is that configurationProvider.get(event) will never return null. If it does, is because a regression was introduced at some point.